endpoints/events/vtrack.go

```diff
@@ -264,10 +269,11 @@ func getBiddersAllowingVastUpdate(req *BidCacheRequest, bidderInfos *config.Bidd
 }
 
 // isAllowVastForBidder checks if a bidder is active and allowed to modify vast xml data
-func isAllowVastForBidder(bidder string, bidderInfos *config.BidderInfos, allowUnknownBidder bool) bool {
+func isAllowVastForBidder(bidder string, bidderInfos *config.BidderInfos, allowUnknownBidder bool, normalizeBidderName normalizeBidderName) bool {
 	//if bidder is active and isModifyingVastXmlAllowed is true
 	// check if bidder is configured
-	if b, ok := (*bidderInfos)[bidder]; bidderInfos != nil && ok {
+	normalizedBidder, _ := normalizeBidderName(bidder)
```

Comment:
The second return value of `normalizeBidderName` is a boolean that tell us whether or not `bidder` exists in the `coreBidderNames` list. I think we shouldn't ignore it.

Reply:
prior to this commit change. today  `isAllowVastForBidder` function is being setup to handle unknown bidder name using `allowUnknowBidder` bool from input arg. Value for `allowUnknowBidder` comes from config i.e `Cfg.VTrack.AllowUnknownBidder`.

on master:

```
func isAllowVastForBidder(bidder string, bidderInfos *config.BidderInfos, allowUnknownBidder bool, normalizeBidderName normalizeBidderName) bool {
	//if bidder is active and isModifyingVastXmlAllowed is true
	// check if bidder is configured
	if b, ok := (*bidderInfos)[bidder]; bidderInfos != nil && ok {
		// check if bidder is enabled
		return b.IsEnabled() && b.ModifyingVastXmlAllowed
	}

	return allowUnknownBidder
}
```

now with context to this PR changes,  `normalizedBidder` will be empty if bidder doesn't exist and in this case check on line 276 `if b, ok := (*bidderInfos)[normalizedBidder.String()]; bidderInfos != nil && ok {` will fail and then `allowUnknownBidder` will be used to decide whether to allow vast xml for bidder. This is similar to behavior of master's version of isAllowVastForBidder. Therefore I think we can ignore second bool returned from `normalizeBidderName`. Let me know your thoughts

Reply:
I see your point Onkar, but I believe a small optimization could result in checking the boolean value returned by `normalizeBidderName(bidder)` because we could aviod the stack allocation the `String()` function call requires.

Reply:
avoiding stack allocation makes sense. https://github.com/prebid/prebid-server/pull/3196/commits/27aa72b303df87cf6241127583583e41c0ae095e makes the change to check boolean returned from `normalizeBidderName `

@hhhjort  change from 27aa72b303df87cf6241127583583e41c0ae095e  commit dismissed previous approval. Requesting to please take a look